File: android/src/main/java/com/reactnativechimesdk/ChimeVideoViewManager.java
package com.reactnativechimesdk;

import android.content.Context;
import android.view.LayoutInflater;

import androidx.annotation.NonNull;

import com.facebook.react.uimanager.SimpleViewManager;
import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.annotations.ReactProp;
import com.reactnativechimesdk.views.ChimeView;

public class ChimeVideoViewManager extends SimpleViewManager<ChimeView> {

  @NonNull
  @Override
  public String getName() {
    return "ChimeVideoView";
  }

  @NonNull
  @Override
  protected ChimeView createViewInstance(@NonNull ThemedReactContext themedReactContext) {
    LayoutInflater inflater = (LayoutInflater) themedReactContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    return (ChimeView) inflater.inflate(R.layout.video_chime_view, null);
  }

  @ReactProp(name = "userID")
  public void bindVideoTile(ChimeView renderView, String attendeeId) {
    renderView.bindVideoTile(attendeeId);
  }
}

File: android/src/main/java/com/reactnativechimesdk/ChimeVideoViewManager.kt
package com.reactnativechimesdk

import android.annotation.SuppressLint
import android.content.Context
import android.view.LayoutInflater
import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.annotations.ReactProp
import com.reactnativechimesdk.views.ChimeView

class ChimeVideoViewManager : SimpleViewManager<ChimeView>() {

  companion object {
    const val NAME = "ChimeVideoView"
    const val TAG = "SimpleChimeViewLog"
  }

  override fun getName(): String = NAME

  @SuppressLint("InflateParams")
  override fun createViewInstance(reactContext: ThemedReactContext): ChimeView {
    val inflater = reactContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
    return inflater.inflate(R.layout.video_chime_view, null) as ChimeView
  }

  @ReactProp(name = "userID")
  fun bindVideoTile(renderView: ChimeView, attendeeId: String) {
    renderView.bindVideoTile(attendeeId)
  }
}
